### ISSUE TITLE
Unexpected Panics

### ISSUE THREAD

**Issue by rkusa** [paupino/rust-decimal#71]
Hi,

I understand that rust-decimal panics in case of unrecoverable errors. However, Today I encountered two panics that seem odd.

```rust
    let rate = Decimal::new(19, 2);
    let part = rate / (rate + Decimal::new(1, 0));
    let result = Decimal::new(169, 0) * part; // panics
```

This will panic with `Multiplication underflowed`. However if I switch the operands (to `part * Decimal::new(169, 0)`), it does not panic anymore.

I tried to also fix this issue by rounding the `part` variable first, but this panicked as well:

```rust
    let rate = Decimal::new(19, 2);
    let part = rate / (rate + Decimal::new(1, 0));
    let part = part.round_dp(2); // panics
```

It panics with `Cannot have an exponent greater than 28 (29 > 28)`.

Thanks

**Comment by paupino** [comment]
I think this is due to an incorrect assumption I made in terms of underflow vs rounding (i.e. here https://github.com/paupino/rust-decimal/blob/master/src/decimal.rs#L1624). Let me reproduce this in a test and see what I can do to fix it.

**Comment by paupino** [comment]
(i.e. We're not losing precision if they're 0's)

**Comment by paupino** [comment]
I've discovered the bug: it's in the division routine whereby we get an underflow in the division however don't attempt to resolve it there. I understand the problem now so will look to get out a solution as soon as I can. The result SHOULD be scale 9 however because we set underflow to true we keep it in an intermediary form (29). This is a shared issue in both simply because division _silently_ allows invalid scales at the moment.
